crates/oxc_linter/src/rules/eslint/no_duplicate_imports.rs

```diff
@@ -0,0 +1,315 @@
+use std::collections::HashMap;
+
+use oxc_ast::{
+    ast::{ExportNamedDeclaration, ImportDeclaration, ImportDeclarationSpecifier},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule};
+
+fn no_duplicate_imports_diagnostic(module_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("'{}' import is duplicated", module_name))
+        .with_help("Merge the duplicated import into a single import statement")
+        .with_label(span)
+}
+
+fn no_duplicate_exports_diagnostic(module_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("'{}' export is duplicated", module_name))
+        .with_help("Merge the duplicated exports into a single export statement")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoDuplicateImports {
+    include_exports: bool,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow duplicate module imports
+    ///
+    /// ### Why is this bad?
+    /// Using a single import statement per module will make the code clearer because you can see everything being imported from that module on one line.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// import { merge } from 'module';
+    /// import something from 'another-module';
+    /// import { find } from 'module';
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// import { merge, find } from 'module';
+    /// import something from 'another-module';
+    /// ```
+    NoDuplicateImports,
+    nursery,
+    pending);
+
+#[derive(Debug, Clone)]
+enum DeclarationType {
+    Import,
+    Export,
+}
+
+#[derive(Debug, Clone)]
+struct ModuleEntry {
+    declaration_type: DeclarationType,
+}
+
+impl Rule for NoDuplicateImports {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let Some(value) = value.get(0) else { return Self { include_exports: true } };
+        Self {
+            include_exports: value
+                .get("includeExports")
+                .and_then(serde_json::Value::as_bool)
+                .unwrap_or(false),
+        }
+    }
+
+    fn run_once(&self, ctx: &LintContext) {
+        let semantic = ctx.semantic();
+        let nodes = semantic.nodes();
+
+        let mut modules: HashMap<String, Vec<ModuleEntry>> = HashMap::new();
+
+        for node in nodes {
```

Comment:
rather than looping over all nodes here, is it possible to use `ctx.module_record().import_entries`?

it has a vec of `ImportEntry` which includes both the span of the module specifier and it's name

Reply:
It could for imports only. 
Then we need to use ` ctx.module_record().export_default` and `ctx.module_record().star_export_entries` and `ctx.module_record().exported_bindings` and ... so the code became complex. 

What is your mind about this ?


Moreover, I am not sure that we could complete tests cases where exports are after the imports one, for instance :
`r#"export * as os from "os"; import os from "os";"#` AND test cases where "imports are before the exports one.

Reply:
Ok it seems I could try, but some use cases are tricky to fix. 
@camc314 Do you think  we improve performances with this over iterating on all nodes ?

Reply:
I completed to cover all test cases with your proposal

Reply:
apologies, missed your response, looks like youve done a great job 🙂